lib/DxilPIXPasses/PixPassHelpers.cpp

```diff
@@ -450,6 +450,56 @@ void ReplaceAllUsesOfInstructionWithNewValueAndDeleteInstruction(
   delete Instr;
 }
 
+unsigned int FindOrAddSV_Position(hlsl::DxilModule &DM,
+                                  llvm::StringRef UpStream) {
+  hlsl::DxilSignature &InputSignature = DM.GetInputSignature();
+  auto &InputElements = InputSignature.GetElements();
+
+  auto Existing_SV_Position =
+      std::find_if(InputElements.begin(), InputElements.end(),
+                   [](const std::unique_ptr<DxilSignatureElement> &Element) {
+                     return Element->GetSemantic()->GetKind() ==
+                            hlsl::DXIL::SemanticKind::Position;
+                   });
+
+  // SV_Position, if present, has to have full mask, so we needn't worry
+  // about the shader having selected components that don't include x or y.
+  // If not present, we add it.
+  if (Existing_SV_Position == InputElements.end()) {
+
+    unsigned int Index = static_cast<unsigned int>(InputElements.size());
+
+    unsigned int StartRow = Index;
+    unsigned int StartColumn = 0;
+    unsigned int RowCount = 1;
+    unsigned int ColumnCount = 4;
+    if (!UpStream.empty()) {
+      SmallVector<StringRef, 4> indices;
+      UpStream.split(indices, "-");
+      if (indices.size() == 4) {
+        // These are encoded in ./DxilPixReadOutputSig.cpp
+        StartRow = atoi(indices[0].str().c_str());
+        StartColumn = atoi(indices[1].str().c_str());
+        RowCount = atoi(indices[2].str().c_str());
+        ColumnCount = atoi(indices[3].str().c_str());
+      }
+    }
+
+    auto Added_SV_Position =
+        llvm::make_unique<DxilSignatureElement>(DXIL::SigPointKind::PSIn);
+    Added_SV_Position->Initialize("Position", hlsl::CompType::getF32(),
+                                  hlsl::DXIL::InterpolationMode::Linear,
+                                  RowCount, ColumnCount, StartRow, StartColumn);
```

Comment:
Still need to add the ID, and if `RowCount > 1`, then you'll need to append sequentially incrementing index... but how would that be?  If that should never be the case, then why is the option available in `UpStream` that you're parsing?  If `UpStream` is more generic than for `SV_Position`, then we should fail on unexpected scenarios, like `RowCount > 1`.

Reply:
As commented, the ID is added for you by AppendElement...